OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -1088,6 +1089,109 @@ public void ResolveOrder(Actor self, Order order)
 
 		#endregion
 
+		bool CanOrder(Actor actor, CPos cell)
+		{
+			if (actor == null || actor.IsDead)
+				return false;
+
+			var aircraft = actor.TraitOrDefault<Aircraft>();
+			return aircraft != null && (aircraft.Info.MoveIntoShroud || self.Owner.Shroud.IsExplored(cell));
+		}
+
+		public IEnumerable<Pair<Actor, CPos>> GroupMove(CPos cell, Order order)
+		{
+			// Make sure we place the units that are closer to the target first to prevent unnecessary shuffles.
+			var selection = order.ExtraActors.Where(a => CanOrder(a, cell)).OrderBy(a => (a.Location - cell).LengthSquared);
```

Comment:
We query the `Aircraft` trait once per actor inside `CanOrder`, and then twice again for each actor below. We could reduce the trait lookups by mapping `ExtraActors` to `TraitPair<Aircraft>`s and filtering out non-aircraft based on the trait being null.